internal/types/types.go

```diff
@@ -61,3 +64,27 @@ type DisplayPath string
 func (u DisplayPath) Empty() bool {
 	return len(u) == 0
 }
+
+// FileSystemPath represents a FileSystem and Path pair. The Path is always valid on the
+// associated FileSystem, but is not guaranteed to be meaningful on the os disk or other
+// unrelated FileSystem instances.
+type FileSystemPath = paths.FileSystemPath
```

Comment:
Please consider renaming this. This is going to be crucial as it is being used everywhere. Name doesn't suggest that FileSystem is also included in the struct. Some alternatives, PathAndFileSystem, PathWithFileSystem, FileSystemAndPath etc.